sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -55,33 +58,50 @@ public synchronized void addActivity(final @NotNull Activity activity) {
       return;
     }
     frameMetricsAggregator.add(activity);
+    snapshotFrameCountsAtStart(activity);
   }
 
-  @SuppressWarnings("NullAway")
-  public synchronized void setMetrics(
-      final @NotNull Activity activity, final @NotNull SentryId sentryId) {
+  private void snapshotFrameCountsAtStart(final @NotNull Activity activity) {
+    FrameCounts frameCounts = calculateCurrentFrameCounts();
+    if (frameCounts != null) {
+      frameCountAtStartSnapshots.put(activity, frameCounts);
+    }
+  }
+
+  private @Nullable FrameCounts diffFrameCountsAtEnd(final @NotNull Activity activity) {
+    @Nullable
+    final FrameCounts frameCountsAtStartFromMap = frameCountAtStartSnapshots.remove(activity);
+    @NotNull
+    final FrameCounts frameCountsAtStart =
+        frameCountsAtStartFromMap == null ? new FrameCounts(0, 0, 0) : frameCountsAtStartFromMap;
```

Comment:
`m`: When do we run into this edge case that the `frameCountsAtStartFromMap` is `null`? We already check that the activity is not `null` here https://github.com/getsentry/sentry-java/blob/f6b0dc7a17408741396774173c2637b717e3cbe1/sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java#L169-L172
The parameter is also annotated `@NotNull`. If this happens, something is wrong and I think we should better not count the frames at all and return `null` here. If we use `FrameCounts(0, 0, 0)`, I think we risk calculating incorrect frame counts.

Reply:
My idea here was that we might not be able to determine frame counts if the `FrameMetricsAggregator` was freshly created as `getMetrics()` on it is marked with `@Nullable`. Since I changed some code around it shouldn't be necessary as we have a `FrameCounts(0, 0, 0)`. Gonna change it so it doesn't calculate a difference if it can't find a start snapshot.